packages/react-components/src/components/table-output-component.tsx

```diff
@@ -528,7 +538,42 @@ export class TableOutputComponent extends AbstractOutputComponent<TableOutputPro
             }
 
             const currRowIndex = this.gridApi?.getLastDisplayedRow() + 1;
-            const lineIndex = await this.findNextMatchIndex(currRowIndex);
+            const lineIndex = await this.findMatchIndex(currRowIndex);
+            if (lineIndex) {
+                this.gridApi.ensureIndexVisible(lineIndex);
+                this.selectStartIndex = this.selectEndIndex = lineIndex;
+                this.handleRowSelectionChange();
+                this.enableIndexSelection = true;
+                this.selectRows();
+            }
+        }
+    }
+
+    private async findPreviousMatchedEvent() {
+        let isFound = false;
+        if (this.gridApi) {
+            this.gridApi.deselectAll();
+            const rowNodes: RowNode[] = [];
+
+            const currRowIndex = this.selectStartIndex === -1 ? this.gridApi?.getLastDisplayedRow() + 1 : this.selectStartIndex;
+            this.gridApi.forEachNode(rowNode => rowNodes.push(rowNode));
+
+            rowNodes.reverse().forEach(rowNode => {
+                if (rowNode.rowIndex < Math.max(this.selectStartIndex, this.selectEndIndex) && rowNode.data
+                    && rowNode.data['isMatched'] && !isFound) {
+                    this.gridApi?.ensureIndexVisible(rowNode.rowIndex);
+                    this.selectStartIndex = this.selectEndIndex = rowNode.rowIndex;
+                    this.handleRowSelectionChange();
+                    rowNode.setSelected(true);
+                    isFound = true;
+                }
+            });
+
+            if (isFound) {
+                return;
+            }
+
```

Comment:
findPreviousMatchedEvent and findNextMatchedEvent share a lot of things. Can we reuse the logic?

Reply:
I'll keep the tags API because there might be other client implementations than the theia-trace-extension that relies on the server side filtering. 

I'll remove the code for 'isFiltered' flag since the server is updated. I'll also try to minimize the code duplication findPreviousMatchedEvent and findNextMatchedEvent.